### PR TITLE
Fixes text theme colors in navigation headers

### PR DESCRIPTION
**Pull request by burkeshartsis**
## TODO:

- [X] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [X] Closes:
- - https://3.basecamp.com/3926363/buckets/6947926/todos/2141435292
- - https://3.basecamp.com/3926363/buckets/6947926/todos/2141435834
- [X] No new warnings in tests, in storybook, and in-app
- [X] Upload GIF(s) of iOS and Android
- [X] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by burkeshartsis** [comment]
Don't merge this its not 100% working

File: packages/WillowCreekApp/src/tabs/connect/index.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ const ConnectNavigator = createStackNavigator(
 const EnhancedConnect = withTheme(({ theme, ...props }) => ({
   ...props,
   screenProps: {
-    headerTintColor: theme.colors.text.primary,
+    headerTintColor: theme.colors.action.primary,
```
Comment (vinnyjth) [inline packages/WillowCreekApp/src/tabs/connect/index.js]: This should happen in core as well :) We'll need to make this change in multiple places (I can't remember where the other location is off the top of my head) 

Related: https://3.basecamp.com/3926363/buckets/6947926/todos/2141435292/edit?replace=true

Comment (vinnyjth) [reply packages/WillowCreekApp/src/tabs/connect/index.js]: I believe Neeley and I did that while back and couldn't find a better way. A peek at the `react-navigation`codebase I think confirmed that these values are derived locally to the instance of the usage.

Comment (burkeshartsis) [reply packages/WillowCreekApp/src/tabs/connect/index.js]: 😢

Comment (vinnyjth) [reply packages/WillowCreekApp/src/tabs/connect/index.js]: We talked about making a reusable HOC that you could use to set navigation options from theme values but figured that the additional ergonomics weren't worth the added complexity.

**Comment by burkeshartsis** [comment]
I'm just waiting for my Android sim to boot before I hit merge

File: packages/WillowCreekApp/src/tabs/connect/index.js (hunk 1 above)
Comment (burkeshartsis) [reply packages/WillowCreekApp/src/tabs/connect/index.js]: I think I agree.
